src/qililab/digital/circuit_transpiler.py

```diff
@@ -337,14 +324,14 @@ def _gate_element_to_pulse_event(
         pulse_shape = Factory.get(pulse_shape_copy.pop(RUNCARD.NAME))(**pulse_shape_copy)
 
         # handle measurement gates and target qubits for control gates which might have multi-qubit schedules
-        if isinstance(gate, M):
-            qubit = gate.qubits[0]
-        # for couplers we don't need to set the target qubit
-        elif isinstance(bus.targets[0], Coupler):
-            qubit = None
-        # handle control gates, target should be the qubit target of the bus
-        else:
-            qubit = next(target.qubit_index for target in bus.targets if isinstance(target, Qubit))
+        bus = self.digital_compilation_settings.buses[gate_event.bus]
+        qubit = (
+            gate.qubits[0]
+            if isinstance(gate, M)
```

Comment:
I get that this M gate class is comming from qibo but can we do something regarding that? It is not ok to use a class that is a single letter, I propose that at least we change the name of such imported variables like with from qibo.gates import M as Measure_gate or something

Reply:
I changed the import to `gates.M`.